src/checks/jsonata-syntax-validator.ts

```diff
@@ -0,0 +1,78 @@
+import jsonata from "jsonata";
+import {
+  StateMachine,
+  StateMachineError,
+  StateMachineErrorCode,
+} from "../types";
+import { getStates } from "./get-states";
+
+type Value = string | number | boolean | Record<string, unknown>;
+
+function checkField(
+  stateName: string,
+  fieldName: string,
+  value: Value
+): StateMachineError[] {
+  if (typeof value === "object")
```

Comment:
if you don't mind, with the multiple indentations I think it might be safer to use brackets around if/else